app/controllers/course-overview.ts

```diff
@@ -24,6 +24,14 @@ export default class CourseOverviewController extends Controller {
     }
   }
 
+  get completedStages() {
+    if (!this.model.course) {
+      return [];
+    }
+
+    return this.userRepositories.filter((repository) => repository.course === this.model.course).flatMap((repository) => repository.completedStages);
```

Comment:
```suggestion
    return this.userRepositories.flatMap((repository) => repository.completedStages);
```

The `userRepositories` getter already filters by course, so don't think we need the extra filter